rustworkx-core/src/generators/random_graph.rs

```diff
@@ -900,6 +1063,14 @@ mod tests {
         assert_eq!(g.edge_count(), 189);
     }
 
+    #[test]
+    fn test_random_regular_graph(){
```

Comment:
For this specific graph, add an assertion looping through all nodes and making sure they have the same degree i.e. 2